### PR TITLE
Validate Nric Checksums

### LINKED ISSUE #254: [BUG] NRIC validation
We need to have NRIC validation, the checksum of NRIC needs to be accurate.

[OPTIONAL] Doctor is identified as licence number instead but it is optional.

### PR DESCRIPTION
**Pull request by TheRealRavager**
As in title. Fixes #254

### CODE REVIEW

**Comment by WilliamRyank** [comment]
Can explain the checksum thingy?

**Comment by TheRealRavager** [comment]
> Can explain the checksum thingy?

referenced a website where i got the algo from

File: src/main/java/organice/model/person/Nric.java [hunk 1]
```diff
@@ -32,10 +59,47 @@ public Nric(String nric) {
...
+     * Returns true if a given string is a valid Nric
      */
     public static boolean isValidNric(String test) {
-        return test.toUpperCase().matches(VALIDATION_REGEX);
+        requireNonNull(test);
+        if (!test.toUpperCase().matches(VALIDATION_REGEX)) {
+            return false;
+        }
+
+        return test.toUpperCase().charAt(test.length() - 1) == calculateChecksumLetter(test);
```
Comment (WilliamRyank) [inline src/main/java/organice/model/person/Nric.java]: tbh i prefer a method that give me boolean `isValidChecksum` that takes in nric and return true or false, so you can just use `test = test.toUpperCase() -> return test.matches(VALIDATION_REGEX) && isValidChecksum(test)`? What do you think

Comment (TheRealRavager) [reply src/main/java/organice/model/person/Nric.java]: I would agree but I am also using this method to generate the checksums for `SampleDataUtil`

**Pull request merged**